File: 1444-NumberOfStepsToReduceANumberToZero/1444-NumberOfStepsToReduceANumberToZero.java
// Last updated: 4/4/2026, 11:02:34 PM
class Solution {
    public int numberOfSteps(int num) {
        if(num==0) return 0;
        int count=0;
        while(num!=0){
            int bit=num&1;
            if(bit==1){
                count+=2;
            }else{
                count+=1;
            }num>>=1;
        }
        return count-1;
    }
}

File: 1444-NumberOfStepsToReduceANumberToZero/1444-NumberOfStepsToReduceANumberToZero.cpp
// Last updated: 4/4/2026, 11:02:31 PM
class Solution {
public:
    int numberOfSteps(int num) {
        if(num==0) return 0;
        int cnt=0;
        while(num>0){
            if(num&1) cnt+=2;
            else cnt++;
            num>>=1;
        }
        return cnt-1;
    }
};